src/builtin_string.cpp

```diff
@@ -123,6 +123,18 @@ class arg_iterator_t {
         }
         return string_get_arg_argv(&argidx_, argv_);
     }
+
+    const wcstring *nextstr() {
+        if (string_args_from_stdin(streams_)) {
+            return get_arg_stdin() ? &storage_ : NULL;
+        }
+        if (auto arg = string_get_arg_argv(&argidx_, argv_)) {
+            storage_ = wcstring(arg);
```

Comment:
`storage_.assign(arg);` or even `storage_ = arg;` reuses  the existing allocation, so is a little more efficient. (Wrote this before but GitHub ate it?)